### PR TITLE
fix: speed up csv extractor

### PR DESCRIPTION
**Pull request by Ben-Cheng**
Fixes: #81

### CODE REVIEW

File: topo_processor/metadata/loaders/metadata_loader_imagery_historic.py [hunk 1]
```diff
@@ -10,6 +11,9 @@
 
 class MetadataLoaderImageryHistoric(MetadataLoader):
     name = "loader.imagery.historic"
+    csv_dict = {}
```
Comment (blacha) [inline topo_processor/metadata/loaders/metadata_loader_imagery_historic.py:14]: should type the dictionary

File: topo_processor/metadata/loaders/metadata_loader_imagery_historic.py [hunk 2]
```diff
@@ -19,37 +23,39 @@ def is_applicable(self, item: Item) -> bool:
         return True
 
     def add_metadata(self, item: Item) -> None:
-        file_name = os.path.splitext(os.path.basename(item.path))[0]
+        if not self.is_init:
+            self.read_csv()
+        item_basename = os.path.splitext(os.path.basename(item.path))[0]
+        item_metadata = self.csv_dict[item_basename]
+        properties = {
+            "linz:sufi": item_metadata[self.csv_header.index("sufi")],
```
Comment (blacha) [inline topo_processor/metadata/loaders/metadata_loader_imagery_historic.py:31]: is there a way to store these indexes and not re look them up for every line

Comment (Ben-Cheng) [reply topo_processor/metadata/loaders/metadata_loader_imagery_historic.py:31]: good idea, we will look into that

Comment (Ben-Cheng) [reply topo_processor/metadata/loaders/metadata_loader_imagery_historic.py:31]: investigate parse as dictionary / array in the csv lib

File: topo_processor/metadata/loaders/metadata_loader_imagery_historic.py [hunk 3]
```diff
@@ -19,37 +22,39 @@ def is_applicable(self, item: Item) -> bool:
         return True
 
     def add_metadata(self, item: Item) -> None:
-        file_name = os.path.splitext(os.path.basename(item.path))[0]
+        if not self.is_init:
+            self.read_csv()
+
+        item_path_basename = os.path.splitext(os.path.basename(item.path))[0]
+        item_dict = self.csv_dict[item_path_basename]
```
Comment (blacha) [inline topo_processor/metadata/loaders/metadata_loader_imagery_historic.py:29]: `item_dict` may be null if the file does not exist.

Comment (Ben-Cheng) [reply topo_processor/metadata/loaders/metadata_loader_imagery_historic.py:31]: fixed
```
        if item_path_basename not in self.csv_dict:
            raise Exception(f"{item_path_basename} cannot be found in the csv.")
```

**Review by blacha — APPROVED** [review]

**Pull request merged**